docs/src/userguide/language_basics.rst

```diff
@@ -128,75 +128,91 @@ the declaration in most cases:
                 cdef float *g = [1, 2, 3, 4]
                 cdef float *h = &f
 
-In addition to the basic types, C :keyword:`struct`, :keyword:`union` and :keyword:`enum`
-are supported:
 
-.. tabs::
+.. note::
 
-    .. group-tab:: Pure Python
+    There is also support for giving names to types using the
+    ``ctypedef`` statement or the ``cython.typedef()`` function, e.g.
 
-        .. literalinclude:: ../../examples/userguide/language_basics/struct_union_enum.py
+    .. tabs::
 
-        .. note:: Currently, Pure Python mode does not support enums. (GitHub issue :issue:`4252`)
+        .. group-tab:: Pure Python
 
-    .. group-tab:: Cython
+            .. code-block:: python
 
-        .. literalinclude:: ../../examples/userguide/language_basics/struct_union_enum.pyx
+                 ULong = cython.typedef(cython.ulong)
 
-        See also :ref:`struct-union-enum-styles`
+                 IntPtr = cython.typedef(cython.p_int)
 
-        .. note::
+        .. group-tab:: Cython
 
-            Structs can be declared as ``cdef packed struct``, which has
-            the same effect as the C directive ``#pragma pack(1)``.
+            .. code-block:: cython
 
-        Declaring an enum as ``cpdef`` will create a :pep:`435`-style Python wrapper::
+                ctypedef unsigned long ULong
 
-            cpdef enum CheeseState:
-                hard = 1
-                soft = 2
-                runny = 3
+                ctypedef int* IntPtr
 
-        There is currently no special syntax for defining a constant, but you can use
-        an anonymous :keyword:`enum` declaration for this purpose, for example,::
+.. _structs:
 
-            cdef enum:
-                tons_of_spam = 3
+Structs, Uions, Enums
+---------------------
 
-        .. note::
-            the words ``struct``, ``union`` and ``enum`` are used only when
-            defining a type, not when referring to it. For example, to declare a variable
-            pointing to a ``Grail`` struct, you would write::
+In addition to the basic types, C :keyword:`struct`, :keyword:`union` and :keyword:`enum`
+are supported:
 
-                cdef Grail *gp
+.. tabs::
 
-            and not::
+    .. group-tab:: Pure Python
 
-                cdef struct Grail *gp  # WRONG
+        .. literalinclude:: ../../examples/userguide/language_basics/struct.py
 
-.. note::
+    .. group-tab:: Cython
 
-    There is also support for giving names to types using the
-    ``ctypedef`` statement or the ``cython.typedef()`` function, e.g.
+        .. literalinclude:: ../../examples/userguide/language_basics/struct.pyx
 
-    .. tabs::
+    .. note::
 
-        .. group-tab:: Pure Python
+        Structs can be declared as ``cdef packed struct``, which has
+        the same effect as the C directive ``#pragma pack(1)``.
```

Comment:
This doesn't seem to appear in the generated output.

I also wonder if it should have a little more detail on two points:
* removes the empty space between members that C automatically to ensure that they're aligned in memory (just people that don't know what `#pragma pack(1)` does).
* the main use is that numpy structured arrays store their data in packed form, so a `cdef packed struct` can be used in a memoryview to match that.

Reply:
done